NOTICES.txt

```diff
@@ -360,200 +500,873 @@ TERMS AND CONDITIONS FOR USE, REPRODUCTION, AND DISTRIBUTION
 
-------------------
+   END OF TERMS AND CONDITIONS
 
-The following libraries licensed under the Ruby License are used 
-within this repository:
- 
-* highline - https://rubygems.org/gems/highline 
-* json - https://rubygems.org/gems/json 
+   APPENDIX: How to apply the Apache License to your work.
 
-Ruby License - https://www.ruby-lang.org/en/about/license.txt
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
 
-Ruby is copyrighted free software by Yukihiro Matsumoto <matz@netlab.jp>.
-You can redistribute it and/or modify it under either the terms of the
-2-clause BSDL (see the file BSDL), or the conditions below:
+   Copyright (c) 2020 CyberArk Software Ltd. All rights reserved.
 
-  1. You may make and give away verbatim copies of the source form of the
-     software without restriction, provided that you duplicate all of the
-     original copyright notices and associated disclaimers.
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
 
-  2. You may modify your copy of the software in any way, provided that
-     you do at least ONE of the following:
+       http://www.apache.org/licenses/LICENSE-2.0
 
-       a) place your modifications in the Public Domain or otherwise
-          make them Freely Available, such as by posting said
-	  modifications to Usenet or an equivalent medium, or by allowing
-	  the author to include your modifications in the software.
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
 
-       b) use the modified software only within your corporation or
-          organization.
 
-       c) give non-standard binaries non-standard names, with
-          instructions on where to get the original software distribution.
+--------------- SECTION 2: BSD 3-clause "New" or "Revised" License ----------
 
-       d) make other distribution arrangements with the author.
+BSD 3-clause "New" or "Revised" License is applicable to the following component(s).
+
+>>> https://rubygems.org/gems/puma/versions/3.12.2
+
+BSD 3-Clause License
+
+Copyright (c) 2019, Evan Phoenix. Some code by Zed Shaw, (c) 2005.
```

Comment:
On the version license (https://github.com/puma/puma/blob/v3.12.2/LICENSE), it looks like the formatting is a bit different:
```
Some code copyright (c) 2005, Zed Shaw
Copyright (c) 2011, Evan Phoenix
All rights reserved.
```
Let's try to keep the formatting the same as it appears on the original file.